scripts/tidycensus.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+#retriever
+
+import os
+import re
+import subprocess
+import pandas as pd
+
+try:
+    import rpy2.robjects as ro
+    import rpy2.robjects.packages as rpackages
+    from rpy2.robjects import pandas2ri
+    from rpy2.robjects.conversion import localconverter
+except ImportError:
+    pass
+
+from retriever.lib.models import Table, VectorDataset
+from retriever.lib.templates import Script
+from retriever.lib.defaults import DATA_WRITE_PATH
+
+R_string = "Rscript -e \"install.packages('{package_name}')\""
+pattern = "had non-zero exit status"
+packages = ['rgdal', 'units', 'sf', 'tigris', 'tidycensus']
+file_paths = {}
+
+
+class main(Script):
+
+    def __init__(self, **kwargs):
+        Script.__init__(self, **kwargs)
+        self.title = "Load US Census Boundary and Attribute Data as 'tidyverse' and " \
+                    "'sf'-Ready Data Frames"
+        self.name = "tidycensus"
+        self.retriever_minimum_version = '3.0.0'
+        self.version = '1.0.0'
+        self.ref = "https://github.com/walkerke/tidycensus"
+        self.citation = ""
+        self.licenses = [{"name": "MIT"}]
+        self.description = "An integrated R interface to the decennial US Census and American Community Survey APIs and" \
+                           "the US Census Bureau's geographic boundary files. Allows R users to return Census and ACS data as" \
+                           "tidyverse-ready data frames, and optionally returns a listcolumn with feature geometry for all Census" \
+                           "geographies."
+        self.keywords = ["census", "US", "tidycensus", "R", "geographic boundaries"]
+        self.encoding = 'utf-8'
+
+    def census_api_key(api_key):
+        r_string = '''
+        library(tidycensus)
+        census_api_key("{}", install = TRUE)
+        home <- Sys.getenv("HOME")
+        renv <- file.path(home, ".Renviron")
+        readRenviron(renv)
+        '''.format(api_key)
+
+        ro.r(r_string)
+
+    def install_tidycensus():
+        for package in packages:
+            if not rpackages.isinstalled(package):
+                output = subprocess.getoutput(R_string.format(package_name=package))
+                if bool(re.search(pattern, output)):
+                    print(output)
+                    print(
```

Comment:
Could you comment a sample run when the package is 
1) Not available so it is installed
2) Run it with out an internet connection am curious to see what we get